File: candidate-api/src/main/java/com/nebu/candidate/dto/report/ReportInterviewerDTO.java
package com.nebu.candidate.dto.report;

import java.util.List;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@ToString
@EqualsAndHashCode
public class ReportInterviewerDTO {
	
	private long id;
	
	private String name;
	
	private double hourlyRate;
	
	private double sum;
	
	private List<ReportSessionDTO> sessions;

}

File: candidate-client/src/app/models/ReportInterviewerDTO.ts
import { ReportSessionDTO } from "./ReportSessionDTO";

export class ReportInterviewerDTO {

	id: Number;
	name: String;
	hourlyRate: Number;
	sum: Number;
	sessions: ReportSessionDTO[];

	constructor(obj: any) {

		Object.assign(this, obj);

		let sessions = [];
		for(let session of obj.sessions) {
			let parsedSession = new ReportSessionDTO(session);
			sessions.push(parsedSession);
		}

		this.sessions = sessions;

	}

}